### PR TITLE
fix: support flutter project for idp with social providers

### PR DESCRIPTION
**Pull request by lazpavel**
#### Description of changes
- fix import auth with IDP with social providers for flutter projects

#### Description of how you validated changes
- manual test
- e2e test added

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by sobolk — APPROVED** [review]

File: packages/amplify-e2e-tests/src/__tests__/amplify-app.test.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {
   validateAmplifyPush,
   validateFeatureFlags,
 } from '../amplify-app-helpers/amplify-app-validation';
+import { AmplifyFrontend } from 'amplify-cli-core';
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/amplify-app.test.ts:20]: Does this import need to be updated?

```suggestion
import { AmplifyFrontend } from '@aws-amplify/amplify-cli-core';
```

File: packages/amplify-e2e-tests/src/__tests__/pull-2.test.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  amplifyPushAuth,
+  createNewProjectDir,
+  deleteProject,
+  deleteProjectDir,
+  getAppId,
+  initJSProjectWithProfile,
+  pullProject,
+} from '@aws-amplify/amplify-e2e-core';
+import { createIDPAndUserPoolWithOAuthSettings, getShortId, importIdentityPoolAndUserPool } from '../import-helpers';
+import { AmplifyFrontend } from 'amplify-cli-core';
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/pull-2.test.ts:13]: Does this import need to be updated?

```suggestion
import { AmplifyFrontend } from '@aws-amplify/amplify-cli-core';
```

File: packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts [hunk 3]
```diff
@@ -1,41 +1,43 @@
 import * as fs from 'fs-extra';
 import * as path from 'path';
-import { pathManager } from 'amplify-cli-core';
+import { AmplifyFrontend, pathManager } from 'amplify-cli-core';
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts:3]: Does this import need to be updated?

```suggestion
import { AmplifyFrontend, pathManager } from '@aws-amplify/amplify-cli-core';
```

File: packages/amplify-category-auth/src/provider-utils/awscloudformation/import/index.ts [hunk 4]
```diff
@@ -78,29 +78,29 @@ export const importResource = async (
 };
 
 const printSuccess = (context: $TSContext, authSelections: AuthSelections, userPool: UserPoolType, identityPool?: IdentityPool): void => {
-  context.print.info('');
+  printer.info('');
```
Comment (jhockett) [inline packages/amplify-category-auth/src/provider-utils/awscloudformation/import/index.ts:81]: nit: I prefer `printer.blankLine();` for it's explicitness

File: packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts (hunk 3 above)
Comment (lazpavel) [reply packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts]: oups, had to revert, looks that amplify-cli-core wasn't yet migrated

Comment (lazpavel) [reply packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts]: committed, this looks like it required some additional changes

**Review by Amplifiyer — DISMISSED** [review]
LGTM, there is a new lint error to look at.

File: packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts (hunk 3 above)
Comment (lazpavel) [reply packages/amplify-e2e-tests/src/amplify-app-helpers/amplify-app-validation.ts]: nevermind,looks that we have 2 packages, one with @aws-amplify and one without

**Review by sdstolworthy — APPROVED** [review]

**Review by sdstolworthy — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**